### PR TITLE
fix: check for plugin method collisions

### PR DESCRIPTION
**Pull request by PaulRBerg**
Addresses https://github.com/cantinasec/review-sablier2/issues/1.

### CODE REVIEW

File: src/interfaces/IPRBProxyRegistry.sol [hunk 1]
```diff
@@ -135,13 +141,14 @@ interface IPRBProxyRegistry {
...
-    /// target contract by forwarding the data. It returns the data it gets back, bubbling up any potential revert.
+    /// target contract by forwarding the data. It returns the data it gets back, and bubbles up any potential revert.
     ///
     /// @dev Emits a {DeployProxy} and an {Execute} event.
     ///
     /// Requirements:
     /// - The caller must not have a proxy.
-    /// - All from {PRBProxy.execute}.
+    /// - `target` must be a contract.
+    /// - `target` must not be the registry.
```
Comment (andreivladbrg) [inline src/interfaces/IPRBProxyRegistry.sol:151]: `target == registry` is not explicitly checked

Comment (PaulRBerg) [reply src/interfaces/IPRBProxyRegistry.sol:151]: Good catch.

File: src/PRBProxyRegistry.sol [hunk 2]
```diff
@@ -191,7 +191,16 @@ contract PRBProxyRegistry is IPRBProxyRegistry {
         // Install every method in the list.
         address owner = msg.sender;
         for (uint256 i = 0; i < length;) {
-            _plugins[owner][methodList[i]] = plugin;
+            // Check for collisions.
+            bytes4 method = methodList[i];
+            if (address(_plugins[owner][method]) != address(0)) {
+                revert PRBProxyRegistry_PluginMethodCollision({
+                    currentPlugin: _plugins[owner][method],
```
[the remark this answers is not in the document]
Comment (PaulRBerg) [reply src/PRBProxyRegistry.sol:198]: 1. I have deliberately not pre-declared the variables to reduce the number of lines of code
2. Via IR may take care of this
3. Pre-declared variables lead to a gas optimization _only_ if the transaction reverts. But we shouldn't optimize for reverts - we should optimize for the happy execution paths.

**Comment by PaulRBerg** [comment]
Yanked in favor of https://github.com/PaulRBerg/prb-proxy/pull/129.

**Pull request merged**